dali/python/backend_impl.cc

```diff
@@ -418,17 +418,29 @@ void ExposeTensorList(py::module &m) { // NOLINT
       py::return_value_policy::reference_internal);
 }
 
-static vector<string> GetRegisteredCPUOps() {
-  return CPUOperatorRegistry::Registry().RegisteredNames();
-}
-
-static vector<string> GetRegisteredGPUOps() {
-  return GPUOperatorRegistry::Registry().RegisteredNames();
+static void FilterInternalOps(vector<string>& op_names) {
+  for (auto it = op_names.begin(); it != op_names.end();) {
+    auto& schema = SchemaRegistry::GetSchema(*it);
+    if (schema.IsInternal()) {
+      op_names.erase(it);
+    } else {
+      it++;
+    }
+  }
 }
 
-static vector<string> GetRegisteredMixedOps() {
-  return MixedOperatorRegistry::Registry().RegisteredNames();
+#define GetRegistredOpsFor(OPTYPE)                                                  \
+static vector<string> GetRegistered##OPTYPE##Ops(bool internal_ops = false) {       \
+  vector<string> op_names = OPTYPE##OperatorRegistry::Registry().RegisteredNames(); \
```

Comment:
also we could move the filtering logic to RegisteredNames implementation

Reply:
correct, would be good to have it at OpFactory level

Reply:
Done